common-pages/src/TimeFilter.tsx

```diff
@@ -0,0 +1,715 @@
+                    <Row isHorizontal={!props.isHorizontal}>
+                        <div className={props.isHorizontal ? (props.showQuickSelect? 'col-2': 'col-3'): 'col-6'}>
+                            <Input<ITimeFilter> Record={filter} Field='halfWindowSize' Setter={(r) => {
+                                const centerTime = getMoment(filter.centerTime);
+                                const [startTime, endTime] = getStartEndTime(centerTime, r.halfWindowSize, units[filter.timeWindowUnits]);
+
+                                setFilter(prevFilter => ({
+                                    ...prevFilter,
+                                    windowSize: r.windowSize,
+                                    halfWindowSize: r.halfWindowSize,
+                                    startTime: startTime.format(momentDateFormat + ' ' + momentTimeFormat),
+                                    endTime: endTime.format(momentDateFormat + ' ' + momentTimeFormat)
+                                }));
+                                setActiveQP(-1);
+                            }} Label='' Valid={() => true}
+                                Type='number' />
+                        </div>
+                        <div className={props.isHorizontal ? (props.showQuickSelect? 'col-2': 'col-3'): 'col-6'}>
+                            <Select<ITimeFilter> Record={filter} Label=''
+                                Field='timeWindowUnits'
+                                Setter={(r) => {
+                                    const centerTime = getMoment(filter.centerTime);
+                                    const [startTime, endTime] = getStartEndTime(centerTime, filter.halfWindowSize, units[r.timeWindowUnits]);
+                                    setFilter(prevFilter => ({
+                                        ...prevFilter,
+                                        timeWindowUnits: r.timeWindowUnits,
+                                        startTime: startTime.format(momentDateFormat + ' ' + momentTimeFormat),
+                                        endTime: endTime.format(momentDateFormat + ' ' + momentTimeFormat)
+                                    }));
+                                    setActiveQP(-1);
+                                }}
+                                Options={[
+                                    { Value: '7', Label: 'Year' },
+                                    { Value: '6', Label: 'Month' },
+                                    { Value: '5', Label: 'Week' },
+                                    { Value: '4', Label: 'Day' },
+                                    { Value: '3', Label: 'Hour' },
+                                    { Value: '2', Label: 'Minute' },
+                                    { Value: '1', Label: 'Second' },
+                                    { Value: '0', Label: 'Millisecond' }
+                                ]} />
```

Comment:
the `Value` should be `TimeUnit`

Reply:
If the value is not a number, then the page won't render properly because a function in 'gpa-gemstones/react-forms/select.tsx' uses findIndex() on Options.Value, and outputs a warning if value is not a number, so I left it as a number for now.